controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1286,6 +1320,76 @@ bool master_thread::handle_cmdu_1905_link_metric_response(Socket *sd,
     return true;
 }
 
+bool master_thread::handle_cmdu_1905_ap_metric_response(Socket *sd,
+                                                        ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received AP_METRIC_RESPONSE_MESSAGE, mid=" << std::dec << int(mid);
+
+    //getting reference for ap metric data storage from db
+    auto &ap_metric_data = database.get_ap_metric_data_map(database.get_gw_mac());
+
+    //will hold new ap metric data from Reporting Agent
+    son::node::ap_metrics_data new_ap_metric_data;
+
+    int tlvType;
+    // holds mac address of the reporting agent, used as ap_metric_data key
+    sMacAddr reporting_agent_bssid;
+
+    // parse all tlvs in cmdu and save ap metric data to database
+    while ((tlvType = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
+
+        if (tlvType < 0) {
+            LOG(ERROR) << "getNextTlvType has failed";
+            return false;
+        }
+
+        if (tlvType == int(wfa_map::eTlvTypeMap::TLV_AP_METRIC)) {
+            //parse tx_ap_metric_data
+            auto apMetricData = cmdu_rx.addClass<wfa_map::tlvApMetric>();
```

Comment:
apMetricData and ap_metric_data are two very similar names for two completely different things....